lib/deqm_test_kit/data_requirements.rb

```diff
@@ -0,0 +1,81 @@
+    title 'Data Requirements'
+    description 'Ensure fhir server can respond to the $data-requirements request'
+
+    fhir_client do
+      url :url
+    end
+
+    fhir_client :embedded_client do
+      url 'http://cqf_ruler:8080/cqf-ruler-r4/fhir'
+    end
+
+    PARAMS = {
+      resourceType: 'Parameters',
+      parameter: [{
+        periodStart: '2019-01-01',
+        periodEnd: '2019-12-31'
+
+      }]
+    }.freeze
+
+    # rubocop:disable Metrics/BlockLength
+    test do
+      title 'Check data requirements against expected return'
+      id 'data-requirements-01'
+      description 'Data requirements on the fhir test server match the data requirements of our embedded client'
+      makes_request :data_requirements
+
+      run do
+        # Look for matching measure from cqf-ruler datastore by resource id
+        # TODO: actually pull measure from user input drop down (populated from embedded client)
+        measure_to_test = 'EXM130|7.3.000'
+        measure_identifier, measure_version = measure_to_test.split('|')
+
+        # Search system for measure by identifier and version
+        fhir_search(:measure, params: { name: measure_identifier, version: measure_version }, name: :measure_search)
+        assert_response_status(200)
+        assert_resource_type(:bundle)
+        assert_valid_json(response[:body])
+        measure_bundle = JSON.parse(response[:body])
+        assert measure_bundle['total'].positive?,
+               "Expected to find measure with identifier #{measure_identifier} and version #{measure_version}"
+
+        id = measure_bundle['entry'][0]['resource']['id']
+
+        # Run our data requirements operation on the test client server
+        fhir_operation("Measure/#{id}/$data-requirements", body: PARAMS, name: :data_requirements)
+        library = JSON.parse(response[:body])
+        actual_dr = library['dataRequirement']
+        assert_response_status(200)
+        assert_resource_type(:library)
+        assert_valid_json(response[:body])
+
+        # Search embedded cqf-ruler instance by identifier and version
+        fhir_search(:measure, client: :embedded_client,
+                              params: { name: measure_identifier, version: measure_version }, name: :measure_search)
+        measure_bundle = JSON.parse(response[:body])
+
+        id = measure_bundle['entry'][0]['resource']['id']
+
+        # Run data requirements operation on embedded cqf-ruler instance
+        fhir_operation("Measure/#{id}/$data-requirements", body: PARAMS, client: :embedded_client,
+                                                           name: :data_requirements)
+        expected_library = JSON.parse(response[:body])
+        expected_dr = expected_library['dataRequirement']
+
+        # Ensure both data requirements results libraries are identical
+        assert((expected_dr - actual_dr).blank?,
+               "Client data-requirements is missing expected data requirements for measure #{id}")
```

Comment:
I know this is the same as the previous data_requirement implementation, but the id here is actually the embedded server's measure id rather than the test client's measure id. I would potentially name those ids two different things and use the test client id here since they can actually differ.